### PR TITLE
Fixes redirect loop on abandoned orders

### PR DESCRIPTION
**Pull request by ashfurrow**
This fixes [PURCHASE-516](https://artsyproduct.atlassian.net/browse/PURCHASE-516). If we don't have an artwork ID, we fall back to redirecting to artsy.net.

### CODE REVIEW

File: src/Apps/Order/redirects.tsx [hunk 1]
```diff
@@ -30,7 +31,10 @@ export const confirmRouteExit = (
 export const shouldRedirect = props => {
   const { location, order, params } = props as any
 
-  if (
+  if (order && order.state === "ABANDONED") {
+    const artworkID = get(order, o => o.lineItems.edges[0].node.artwork.id)
+    throw new RedirectException(artworkID ? `/artwork/${artworkID}` : "/")
```
Comment (damassi) [inline src/Apps/Order/redirects.tsx]: Can you add a comment (similar to below) describing this redirect behavior?

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 5f76732fa6010333de0761fbd2433bcc23252d2a

https://deploy-preview-1413--artsy-reaction.netlify.com

File: src/Apps/Order/__tests__/OrderApp.test.tsx [hunk 2]
```diff
@@ -48,17 +48,42 @@ describe("OrderApp routing redirects", () => {
     expect(redirect).toBe(undefined)
   })
 
-  it("redirects to the status route if the order is not pending", async () => {
+  it("redirects to the artwork page if the order is abandoned", async () => {
```
Comment (sweir27) [inline src/Apps/Order/__tests__/OrderApp.test.tsx]: Can we add back a test that fits the original description? 

Maybe one that is in a `SUBMITTED` state that does redirect to `status`?

**Comment by sweir27** [comment]
One comment, but other than that looks 👍.

**Pull request merged**